client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx

```diff
@@ -137,51 +125,42 @@ const GoogleWorkspaceCard: FunctionComponent< EmailProvidersStackedCardProps > =
 					'This discount is only available the first time you purchase a %(googleMailService)s account, any additional mailboxes purchased after that will be at the regular price.',
 					{
 						args: {
-							googleMailService: getGoogleMailServiceFamily(),
+							googleMailService: googleWorkspace.productName,
 						},
 						comment: '%(googleMailService)s can be either "G Suite" or "Google Workspace"',
 					}
 				) }
 			</InfoPopover>
-		</span>
+		</div>
 	) : null;
 
-	const priceBadge = (
-		<div className="google-workspace-card__price-badge">
-			<PromoCardPrice
-				formattedPrice={ monthlyPrice }
-				discount={
-					productIsDiscounted && (
-						<span className="google-workspace-card__discounted-price">
-							{ getMonthlyPrice( gSuiteProduct.sale_cost, currencyCode ) }
-						</span>
-					)
-				}
-				additionalPriceInformation={
-					<span className="google-workspace-card__provider-additional-price-information">
-						{ googleWorkspace.additionalPriceInformation }
-					</span>
-				}
-			/>
-		</div>
+	googleWorkspace.priceBadge = (
+		<PriceBadge
+			additionalPriceInformationComponent={ discount }
+			priceComponent={ priceWithInterval }
+			className={ 'google-workspace-card' }
+		/>
 	);
 
-	const [ googleUsers, setGoolgeUsers ] = useState( newUsers( selectedDomainName ) );
+	const [ googleUsers, setGoogleUsers ] = useState( newUsers( selectedDomainName ) );
 	const [ addingToCart, setAddingToCart ] = useState( false );
```

Comment:
Thought: we should wrap this function to protect against the state update occurring after this component has been unmounted.
```suggestion
	const [ isMounted, setIsMounted ] = useState( null ); 
	const [ addingToCart, setAddingToCart ] = useState( false );
	useEffect( () => {
		setIsMounted( true );
		return () => {
			setIsMounted( false );
		};
	}, [] );
	const setAddingToCartWithMountCheck = ( isAddingToCart: boolean ): void => {
		if ( isMounted ) {
			setAddingToCart( isAddingToCart );
		}
	};
```